Include/internal/pycore_optimizer.h

```diff
@@ -116,8 +116,7 @@ PyAPI_FUNC(void) _Py_Executor_DependsOn(_PyExecutorObject *executor, void *obj);
 // Export for '_testinternalcapi' shared extension.
 PyAPI_FUNC(_PyOptimizerObject *) _Py_GetOptimizer(void);
 PyAPI_FUNC(int) _Py_SetTier2Optimizer(_PyOptimizerObject* optimizer);
-PyAPI_FUNC(PyObject *) _PyOptimizer_NewCounter(void);
-PyAPI_FUNC(PyObject *) _PyOptimizer_NewUOpOptimizer(void);
```

Comment:
In theory, the only change you need is to remove the `PyAPI_FUNC`. That will stop the optimizer from being exported as public API.

Reply:
If the header file isn't used elsewhere, we can remove it and keep the signature only in the source file.
(Sorry for the term mobile. I don't look it.)

Reply:
`pycore_optimizer.h` is used. Please don't remove it.

Reply:
These 2 APIs are used for `_testinternalcapi` for testing. It seems like `_testinternalcapi` fails to be imported due to missing symbols after removing the `PyAPI_FUNC`.

Reply:
You should likely mark them as `extern` if they are needed to be exported then.

Reply:
Oh, right, it seems that declaring them as extern is not sufficient. So we should probably keep `PyAPI_FUNC`. However, you should add a comment "// Export for '_testinternalcapi' shared extension.".